operator/v1/types_console.go

```diff
@@ -146,6 +146,22 @@ type ConsoleCustomization struct {
 	// +listMapKey=id
 	// +optional
 	Perspectives []Perspective `json:"perspectives"`
+	// consoleURL is an URL to be used as the base console address.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Invalid value will prevent a console rollout.
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
```

Comment:
In general we have been trying to move away from regex validation instead using CEL (kubebuilder:validation:XValidation:rule). perhaps `startsWith()` and a clear message indicating the failure reason?

Reply:
Replaced regexp with CEL validation.

Reply:
Finally [the url library was used](https://github.com/openshift/api/pull/1887#discussion_r1601337866) to do the check for https scheme.